### ISSUE TITLE
Integration with SCT and impossibility to perform DEG on RNA assay

### ISSUE THREAD

**Issue by Behind42** [satijalab/seurat#9424]
Dear all,

I was not aware of the v5 of Seurat and the big changes with layers.. 
So unfortunately I integrated using the SCT pipeline as usual and I identified my clusters etc. 
Then the issue is that even if I join the layers, I can't perform any DEG on the RNA assay, the error is saying that my data slot is empty. 

Do you think I have to redo everything from the begining?

Thank you very much

Quentin

**Comment by cyrillustan** [comment]
Yes, the transition from Seurat v4 to v5 introduces a new paradigm with layers. You do not need to redo everything, but you need to ensure that you are referencing the correct layer for your DEG. SCT should run on "counts" layer and update "data" and "scale.data" layers. Can you check what they are after you run SCT?
